src/main/java/bl4ckscor3/mod/scserverutils/SCServerUtils.java

```diff
@@ -87,4 +109,39 @@ public static void onPlayerSpawnPhantoms(PlayerSpawnPhantomsEvent event) {
 				event.setPhantomsToSpawn(player.level().random.nextIntBetweenInclusive(phantomSpawns.min().get(), phantomSpawns.max().get()));
 		}
 	}
+
+	public static List<ServerPlayer> getOnlineOperators() {
+		MinecraftServer server = ServerLifecycleHooks.getCurrentServer();
+
+		List<ServerPlayer> onlinePlayers = server.getPlayerList().getPlayers();
+
+		ServerOpList opList = server.getPlayerList().getOps();
+
+		return onlinePlayers.stream()
+				.filter(player -> {
+					ServerOpListEntry entry = opList.get(player.getGameProfile());
+					return entry != null && entry.getLevel() >= 1; // niveau OP ≥ 1
+				})
+				.collect(Collectors.toList());
+	}
+
+	public static void logCancelledMessage(ServerPlayer player, String message) {
+		final String logMessage = "[CANCELLED] <" + player.getDisplayName().getString() + "> " + message;
+		LOGGER.info(logMessage);
+
+		for (ServerPlayer p: getOnlineOperators()) {
+			p.sendSystemMessage(Component.literal(logMessage));
+		}
+	}
```

Comment:
These two can probably just go in MuteEventsHandler, as they are not used anywhere else in the mod.